src/tests/UnitTests/Navigation/WPF/NavigationServiceTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿using CodeMonkeys.DependencyInjection;
+using CodeMonkeys.Navigation;
+using CodeMonkeys.Navigation.WPF;
+using CodeMonkeys.UnitTests.Navigation.ViewModels;
+using CodeMonkeys.UnitTests.Navigation.WPF.Views;
+
+using Moq;
+using NUnit.Framework;
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CodeMonkeys.UnitTests.Navigation.WPF
+{
+    public partial class NavigationServiceTests
+    {
+        private readonly Mock<IDependencyContainer> _dependencyContainerMock =
+            new Mock<IDependencyContainer>();
+
+        private readonly SecondViewModel _secondViewModel=
+            new SecondViewModel();
+
+
+        private CodeMonkeys.Navigation.WPF.INavigationService navigationService;
+
+
+
+        [SetUp]
+        public void Setup()
+        {
+            navigationService = new NavigationService(
+                _dependencyContainerMock.Object);
+
+
+            _dependencyContainerMock
+                .Setup(container => container.Resolve<MainViewModel>())
+                .Returns(new MainViewModel());
+
+            _dependencyContainerMock
+                .Setup(container => container.Resolve<SecondViewModel>())
+                .Returns(_secondViewModel);
+
+            _dependencyContainerMock
+                .Setup(container => container.Resolve<ParameterViewModel>())
+                .Returns(new ParameterViewModel());
+        }
+
+        [TearDown]
+        public void Clean()
+        {
+            NavigationService.Configuration.AllowDifferentViewTypeRegistrationForSameViewModel = false;
+
+            navigationService.ResetRegistrations();
+        }
+
+
+
+        [Test, Apartment(ApartmentState.STA)]
+        public async Task SetRootAsync_IfViewModelIsRegistered_RootViewModelAndCurrentViewModelAreSet()
+        {
+            var expectedRootViewModelType = typeof(MainViewModel);
+            navigationService.Register<MainViewModel, MainPage>();
+
+
+            await navigationService.SetRootAsync<MainViewModel>();
+
+
+            Assert.AreEqual(
+                expectedRootViewModelType,
+                navigationService.RootViewModel.GetType());
+
+            Assert.AreEqual(
+                expectedRootViewModelType,
+                navigationService.CurrentViewModel.GetType());
+        }
+
+
+        private void RegisterViewModels()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'RegisterViewModels'.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5575746)